src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSBucketDelete.java

```
@@ -64,11 +65,21 @@ public void run(ActionContext context) throws Exception {
     config.validate(context.getFailureCollector());
 
     Configuration configuration = new Configuration();
-    String serviceAccountFilePath = config.getServiceAccountFilePath();
-    Credentials credentials = serviceAccountFilePath == null ?
-      null : GCPUtils.loadServiceAccountCredentials(serviceAccountFilePath);
-    if (serviceAccountFilePath != null) {
-      configuration.set("google.cloud.auth.service.account.json.keyfile", serviceAccountFilePath);
+
+    Boolean isServiceAccountFilePath = config.isServiceAccountFilePath();
+    if (isServiceAccountFilePath == null) {
+      context.getFailureCollector().addFailure("Service account type is undefined.",
```

Check that service account type is defined in config.validate(). If there are no macros, we should we able to fail on invalid input in `configurePipeline`